### PR TITLE
Add ODBC guide

### PR DESCRIPTION
**Pull request by maiadegraaf**
This PR introduces a guide on ODBC and how to create an C++ application to connect to our driver.

### CODE REVIEW

**Comment by szarnyasg** [comment]
Funnily enough, my VS Code's Markdown plug-in kept regenerating the table of contents (or rather, what it perceived as the TOC under `Setting up an Application`) upon every save operation. I turned this off using:
https://github.com/yzhang-gh/vscode-markdown/issues/388

**Comment by szarnyasg** [comment]
I did a pass, it looks good. Three comments:

1. The `<summary>` tag doesn't work and it should be avoided along with `<details>`. Please remove these.
Screenshot 2023-09-20 at 13 54 19

2. At the beginning of the `Setting up an Application` section, it would be nice to have a sentence on what we are going to achieve in the sections that follow.

3. We have a lot of ODBC docs at https://duckdb.org/docs/api/odbc/overview. These are of varying quality and likely outdated. Could you take a look at them after this PR? (We can merge this first.)

**Review by szarnyasg — CHANGES_REQUESTED** [review]
See https://github.com/duckdb/duckdb-web/pull/1192#issuecomment-1727599294

File: docs/guides/odbc/general.md [hunk 1]
```diff
@@ -39,23 +39,16 @@ The first step is to connect to the data source so that the application can perf
...
+
+| Key        | Value                                                                                                                    |
+|------------|--------------------------------------------------------------------------------------------------------------------------|
+| `DSN`      | A DSN is a string that identifies a database.                                                                            |
+| `FILEDSN`  | The path to a file containing a DSN.                                                                                     |
+| `DRIVER`   | The name of the ODBC driver to use.  This is the name of the driver as it appears in the ODBC Data Source Administrator. |
+| `UID`      | The username to use when connecting to the database.                                                                     |
+| `PWD`      | The password to use when connecting to the database.                                                                     |
+| `SAVEFILE` | The path to a file to which the connection string should be saved.                                                       |
```
[the remark this answers is not in the document]
Comment (szarnyasg) [reply docs/guides/odbc/general.md]: Got it! I put it back to a comment.

**Comment by maiadegraaf** [comment]
>     2. At the beginning of the `Setting up an Application` section, it would be nice to have a sentence on what we are going to achieve in the sections that follow.

I did have something like this before, but it got lost somewhere in the move, added it back in now :)

**Review by szarnyasg — APPROVED** [review]
Looks great, thanks. Pushed to nit-fixing commits and will merge this now.

**Pull request merged**